Code/Framework/AzTest/AzTest/Platform/Linux/platform_linux.cmake

```diff
@@ -6,8 +6,18 @@
 #
 #
 
-# Platform specific cmake file for configuring target compiler/link properties
-# based on the active platform
-# NOTE: functions in cmake are global, therefore adding functions to this file
-# is being avoided to prevent overriding functions declared in other targets platfrom
-# specific cmake files
+# Special case for unit tests that are specific to ubuntu and their versions.
+if ("${LY_LINUX_DISTRO_NAME}" STREQUAL "ubuntu")
+
+	if ("${LY_LINUX_DISTRO_VERSION}" STREQUAL "20.04")
+
+		set(LY_COMPILE_DEFINITIONS PUBLIC LY_UBUNTU_VERSION=2004)
+
+	elseif ("${LY_LINUX_DISTRO_VERSION}" STREQUAL "22.04")
+
+		set(LY_COMPILE_DEFINITIONS PUBLIC LY_UBUNTU_VERSION=2204)
+
+	endif()
+
+endif()
```

Comment:
I am not OK with checking for a specific distro in order to workaround a test issue.
This is not future facing and mayneed to be updated again when `Ubuntu 24.04 LTS` comes out.

It also may not work on other Linux distros, like ArchLinux, Fedora, Debian, etc...

Reply:
We could mock the [PerlinImprovedNoise::PrepareTable](https://github.com/o3de/o3de/blob/development/Gems/GradientSignal/Code/Source/PerlinImprovedNoise.cpp#L136-L147) function to return a fixed table of random values instead.

Reply:
Thats a better suggestion! It appears that std::mt19937 may be deterministic, but only per platform. 

```void PerlinImprovedNoise::PrepareTable(int seed)
    {
        AZStd::array<int, 256> randtable;
        std::iota(randtable.begin(), randtable.end(), 0);
        std::shuffle(randtable.begin(), randtable.end(), std::mt19937(seed));

        for (int x = 0; x < 256; ++x)
        {
            m_permutationTable[x] = randtable[x];
            m_permutationTable[x + 256] = randtable[x];
        }
    }
```

Reply:
Re-did the branch to mock the relevant necessary classes, with as much minimal changes as possible to do so (see https://github.com/o3de/o3de/pull/11723/files#diff-4b98749e4581ce72165659d27b3b0259ec43730ed9d73ced1891e123704b5035 )

Reply:
Update the changes again to support injecting a (fixed) permutation table to eliminate std::mt19937 inconsistencies across platforms (and an explanation)